plugins/nf-wave/src/main/io/seqera/wave/plugin/config/TowerConfig.groovy

```diff
@@ -49,8 +51,12 @@ class TowerConfig {
 
     private String endpoint0(Map opts, Map<String,String> env) {
         def result = opts.endpoint as String
-        if( !result || result=='-' )
-            result = env.get('TOWER_API_ENDPOINT') ?: 'https://api.tower.nf'
+        if( !result || result=='-' ) {
+            String env_api_endpoint = env.get('TOWER_API_ENDPOINT')
+            if( env_api_endpoint && env_api_endpoint.equals('https://api.tower.nf'))
+                log.warn "The endpoint `https://api.tower.nf` is deprecated. Please use `https://api.cloud.seqera.io` instead."
+            result = env_api_endpoint ?: 'https://api.cloud.seqera.io'
+        }
```

Comment:
Instead of changing this, it could be easier just to "catch" the old endpoint when creating the client 

https://github.com/nextflow-io/nextflow/blob/2ee4d11e33778b0b1ebe3f66ad3ef7ecc7a58a80/plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerClient.groovy#L149-L149